instancenorm: eager static quant support

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39413 graph mode: util for fusion of functions which require observation
* #39203 quant layer/group/instance norm: make weights and biases optional
* #39096 instancenorm: static quant graph mode support
* #39095 groupnorm: graph mode static quant support
* #39094 layernorm: eager mode qat support
* #39093 instancenorm: eager mode QAT support
* #39092 groupnorm: eager mode QAT support
* **#39091 instancenorm: eager static quant support**
* #39090 groupnorm: eager static quant support
* #37802 quant: clean up normalization channels_last handling

Summary:

Adds eager mode static quant support for instancenorm.

Test Plan:

```
python test/test_quantization.py TestPostTrainingStatic.test_normalization
python test/test_quantization.py TestStaticQuantizedModule.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21885265](https://our.internmc.facebook.com/intern/diff/D21885265)

vkuzo committed:
instancenorm: eager static quant support

Summary:

Adds eager mode static quant support for instancenorm.

Test Plan:

```
python test/test_quantization.py TestPostTrainingStatic.test_normalization
python test/test_quantization.py TestStaticQuantizedModule.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/nn/quantized/modules/normalization.py:
```diff
@@ -153,3 +153,279 @@ def from_float(cls, mod):
         return new_mod
+
+class InstanceNorm1d(torch.nn.InstanceNorm1d):
+    r"""Applies Instance Normalization over a 3D input (a mini-batch of 1D
```
are these copied from floating point version? should we link to the fp version instead of copying?

vkuzo replied:
to clarify, you mean the docblock?

jerryzh168 replied:
Yeah I mean the docs

supriyar commented on torch/testing/_internal/common_quantization.py:
```diff
@@ -408,12 +408,19 @@ def __init__(self):
         self.fc1 = torch.nn.Linear(5, 8).to(dtype=torch.float)
         self.layer_norm = torch.nn.LayerNorm((8))
         self.group_norm = torch.nn.GroupNorm(2, 8)
+        # TODO: add handling for affine=False (future PR)
```
For my understanding, why is affine=False different?

vkuzo replied:
it's a bug as of this revision, fixed it further upstream in https://github.com/pytorch/pytorch/pull/39203

vkuzo committed:
Update on "instancenorm: eager static quant support"


Summary:

Adds eager mode static quant support for instancenorm.

Test Plan:

```
python test/test_quantization.py TestPostTrainingStatic.test_normalization
python test/test_quantization.py TestStaticQuantizedModule.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar reviewed (approved):
LG

vkuzo committed:
Update on "instancenorm: eager static quant support"


Summary:

Adds eager mode static quant support for instancenorm.

Test Plan:

```
python test/test_quantization.py TestPostTrainingStatic.test_normalization
python test/test_quantization.py TestStaticQuantizedModule.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "instancenorm: eager static quant support"


Summary:

Adds eager mode static quant support for instancenorm.

Test Plan:

```
python test/test_quantization.py TestPostTrainingStatic.test_normalization
python test/test_quantization.py TestStaticQuantizedModule.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21885265](https://our.internmc.facebook.com/intern/diff/D21885265)

[ghstack-poisoned]

vkuzo committed:
Update on "instancenorm: eager static quant support"


Summary:

Adds eager mode static quant support for instancenorm.

Test Plan:

```
python test/test_quantization.py TestPostTrainingStatic.test_normalization
python test/test_quantization.py TestStaticQuantizedModule.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21885265](https://our.internmc.facebook.com/intern/diff/D21885265)

[ghstack-poisoned]

vkuzo committed:
Update on "instancenorm: eager static quant support"


Summary:

Adds eager mode static quant support for instancenorm.

Test Plan:

```
python test/test_quantization.py TestPostTrainingStatic.test_normalization
python test/test_quantization.py TestStaticQuantizedModule.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21885265](https://our.internmc.facebook.com/intern/diff/D21885265)

[ghstack-poisoned]

vkuzo committed:
Update on "instancenorm: eager static quant support"


Summary:

Adds eager mode static quant support for instancenorm.

Test Plan:

```
python test/test_quantization.py TestPostTrainingStatic.test_normalization
python test/test_quantization.py TestStaticQuantizedModule.test_instance_norm
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21885265](https://our.internmc.facebook.com/intern/diff/D21885265)

[ghstack-poisoned]
